mesonbuild/interpreter.py

```diff
@@ -2786,6 +2815,10 @@ def check_subproject_version(wanted, found):
 
     def get_subproject_dep(self, name, dirname, varname, required):
         try:
+            subproject = self.subprojects[dirname]
+            if is_disabler(subproject):
+                return subproject
+
```

Comment:
I would use `if not subproject.found():` that's more explicit.

If subproject is not found and required arg is True you need to raise exception, if required is False you should return None and log a message.

Reply:
Using `.found()` instead.